profiler/profiler_test.go

```diff
@@ -82,6 +82,20 @@ func TestStart(t *testing.T) {
 		mu.Unlock()
 	})
 
+	t.Run("dd_profiling_not_enabled", func(t *testing.T) {
+		t.Setenv("DD_PROFILING_ENABLED", "false")
+		if err := Start(); err != nil {
+			t.Fatal(err)
+		}
+		defer Stop()
+
+		mu.Lock()
+		// if DD_PROFILING_ENABLED is false, the profiler should not be started even if Start() is called
+		// So we should not have an activeProfiler
+		assert.Nil(t, activeProfiler)
+		mu.Unlock()
+	})
```

Comment:
Thanks for adding this and the other test case.

NIT: Some of the existing test suite has a lot of tests like this that assert on the internal state of things. However, in general we prefer tests that verify user-visible behavior these days. I.e. having a test that checks that a disabled profiler doesn't send data would be nice. In practice this might be a difficult test to write in a non-flaky manner, so I'm okay with keeping the testing as proposed in this PR.

Reply:
If I understand correctly, should we keep the test that checks activeProfiler, and additionally create a test that verifies no profiling data is sent when `DD_PROFILER_ENABLED=false`?
As you mentioned, since I'm not very familiar with the codebase, implementing this might be a bit challenging, but I'll give it a try. Thank you for the suggestion.

Reply:
Here's a test for `DD_PROFILING_ENABLED=false` I sketched up while reviewing this:

```
func TestEnabledFalse(t *testing.T) {
        t.Setenv("DD_PROFILING_ENABLED", "false")
        ch := startTestProfiler(t, 1, WithPeriod(10*time.Millisecond), WithProfileTypes())
        select {
        case <-ch:
                t.Fatal("received profile when profiler should have been disabled")
        case <-time.After(time.Second):
                // This test might succeed incorrectly on an overloaded
                // CI server, but is very likely to fail locally given a
                // buggy implementation
        }
}
```

Feel free to add it to the PR if it makes sense to you. We can keep the other tests for now.

Reply:
I've implemented additional changes in [f71e851 (#2840)](https://github.com/DataDog/dd-trace-go/pull/2840/commits/f71e851689c77503416e805091186535e7e0dd27). 

However, the tests you've already suggested seem sufficient, so I couldn't find any points to modify or add. If there are any areas that need improvement, please feel free to let me know!